### DISCUSSION TITLE
Repository Groups update - now the same repo can belong to multiple repository groups

### DISCUSSION THREAD

**Discussion by lukaszgryglicki** [cncf/devstats#32] [Announcements]
This allows more flexibility on repository groups.
- One repository can belong to multiple repository groups.
- One repository group can have multiple repositories (this was supported since the begginning).

This allows for example adding "GitHub org wide" repository groups by default.
Many CNCF projects used "default" repository group selections which was just the newest name for a given repo is a single repository group (repositories could have more historical names) - so by default it was almost 1:1 mapping (repo current name = repository group). Some projects had custom definitions like Kuberentes (SIGs and few others), few more CNCF projects + the spcial `All CNCF` instance had repo group = a single CNCF project.

Now all projects that used default mapping (repo group = the final repo name) additinally have "GitHub orgg wide" repository groups, that means for example if a given project have 10 repos under GitHub org A and 20 repos under GitHub orb B - there will be 32 repo groups.
10 + 20 =30 single repo ones and additionally A (having 10 repos) and B (having 20 repos).
So for example repo "A/a" will belong to "A/a" repo group and "A" repo group.

This allows by-default org-wide stats calculations.

I really like this, it was originall requested [here](https://github.com/cncf/devstats/issues/28), and I added support for this in other dashboards and projects as well.

cc @caniszczyk 

BTW: this is not yet “populated” on every single dashboard & project, but tested and populated on some, and they use shared metrics, so eventually it will be everywhere, I think it is good time to announce.

There are 40+ dashboards for all 150+ projects (not all of them use repo groups) so it might take some time… but will eventually come, I’m now monitoring this.

But I still wanted to `share` this big update IMHO.

**Comment by caniszczyk** [comment]
LGTM


On Fri, Sep 29, 2023 at 10:18 AM Łukasz Gryglicki ***@***.***>
wrote:

> This allows more flexibility on repository groups.
>
>    - One repository can belong to multiple repository groups.
>    - One repository group can have multiple repositories (this was
>    supported since the begginning).
>
> This allows for example adding "GitHub org wide" repository groups by
> default.
> Many CNCF projects used "default" repository group selections which was
> just the newest name for a given repo is a single repository group
> (repositories could have more historical names) - so by default it was
> almost 1:1 mapping (repo current name = repository group). Some projects
> had custom definitions like Kuberentes (SIGs and few others), few more CNCF
> projects + the spcial All CNCF instance had repo group = a single CNCF
> project.
>
> Now all projects that used default mapping (repo group = the final repo
> name) additinally have "GitHub orgg wide" repository groups, that means for
> example if a given project have 10 repos under GitHub org A and 20 repos
> under GitHub orb B - there will be 32 repo groups.
> 10 + 20 =30 single repo ones and additionally A (having 10 repos) and B
> (having 20 repos).
> So for example repo "A/a" will belong to "A/a" repo group and "A" repo
> group.
>
> This allows by-default org-wide stats calculations.
>
> I really like this, it was originall requested here
> <https://github.com/cncf/devstats/issues/28>, and I added support for
> this in other dashboards and projects as well.
>
> cc @caniszczyk <https://github.com/caniszczyk>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cncf/devstats/discussions/32>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAPSINASD7MACFY2OPCEW3X43REJANCNFSM6AAAAAA5MT7GQ4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Cheers,

Chris Aniszczyk
https://aniszczyk.org

**Comment by lukaszgryglicki** [comment]
Implemented in all projects. Closing.
